gpflow/base.py

```diff
@@ -51,15 +51,16 @@ def __init__(self,
         """
         super().__init__()
 
-        value = _verified_value(value, dtype)
+        self._transform = transform
```

Comment:
Could you move it back to the end? The reason is that attribute assign should happen after all potential failure cases.

Reply:
No I can't, I moved it to the front so that it's ready for validate_unconstrained_value() before the Variable construction.

Reply:
Ok, then this is a separate issue, dependencies in the constructor calls and existing functions.